### PR TITLE
Laboratory work #3, Elena Chevtaeva - 23FPL2

### PR DESCRIPTION
**Pull request by Helenhelen2005**

### CODE REVIEW

File: lab_3_ann_retriever/main.py [hunk 1]
```diff
@@ -130,6 +152,18 @@ def _remove_stop_words(self, tokens: list[str]) -> list[str] | None:
...
+        for token in tokens:
+            if not isinstance(token, str):
+                return None
+        for word in self.stop_words:
+            if not isinstance(word, str):
+                return None
+        if len(tokens) == 0 or len(self.stop_words) == 0:
+            return None
+
+        return list(filter(lambda x: x not in self.stop_words, tokens))
```
Comment (Vasilisa-Blyudova) [inline lab_3_ann_retriever/main.py:166]: great!

File: lab_3_ann_retriever/main.py [hunk 2]
```diff
@@ -104,6 +105,17 @@ def tokenize(self, text: str) -> list[str] | None:
...
         """
+        if not isinstance(text, str) or len(text) < 1:
+            return None
+
+        for symbol in text:
+            if symbol.isalpha() or symbol == " ":
+                continue
+            text = text.replace(symbol, " ")
+
+        processed_text = self._remove_stop_words(text.lower().split())
```
Comment (Vasilisa-Blyudova) [inline lab_3_ann_retriever/main.py:116]: don't create separate variable

**Review by Vasilisa-Blyudova — CHANGES_REQUESTED** [review]

**Comment by Vasilisa-Blyudova** [comment]
No change since last review

File: lab_3_ann_retriever/main.py (hunk 2 above)
Comment (Helenhelen2005) [reply lab_3_ann_retriever/main.py]: fixed

File: lab_3_ann_retriever/main.py [hunk 3]
```diff
@@ -49,6 +52,19 @@ def calculate_distance(query_vector: Vector, document_vector: Vector) -> float |
 
     In case of corrupt input arguments, None is returned.
     """
+    bad_input = (query_vector is None or document_vector is None or
```
Comment (artyomtugaryov) [inline lab_3_ann_retriever/main.py:55]: too much

File: lab_3_ann_retriever/main.py [hunk 4]
```diff
@@ -149,6 +197,11 @@ def __init__(self, corpus: list[list[str]]) -> None:
         Args:
             corpus (list[list[str]]): Tokenized documents to vectorize
         """
+        self._corpus = corpus
+        self._idf_values = {}
+        self._vocabulary = []
+        self._token2ind = {}
+        self.build()
```
Comment (artyomtugaryov) [inline lab_3_ann_retriever/main.py:204]: really? do you need this?

File: lab_3_ann_retriever/start.py [hunk 5]
```diff
@@ -30,7 +32,21 @@ def main() -> None:
     """
     with open("assets/secrets/secret_1.txt", "r", encoding="utf-8") as text_file:
         text = text_file.read()
-    result = None
+
+    documents = open_files()[0]
```
Comment (artyomtugaryov) [inline lab_3_ann_retriever/start.py:36]: do not call this function twice; return it from the one call

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_3_ann_retriever/main.py (hunk 4 above)
Comment (Helenhelen2005) [reply lab_3_ann_retriever/main.py]: removed

File: lab_3_ann_retriever/start.py (hunk 5 above)
Comment (Helenhelen2005) [reply lab_3_ann_retriever/start.py]: done

File: lab_3_ann_retriever/main.py (hunk 3 above)
Comment (Helenhelen2005) [reply lab_3_ann_retriever/main.py]: fixed

**Pull request closed without merging**